python/ray/data/dataset.py

```diff
@@ -1701,24 +1701,40 @@ def union(self, *other: List["Dataset"]) -> "Dataset":
         )
 
     def groupby(self, key: Optional[str]) -> "GroupedData":
-        """Group the dataset by the key function or column name.
+        """Group rows of a :class:`Dataset` according to a column.
+
+        This method is useful if you want to transform data differently depending on a
+        categorical variable.
 
         Examples:
-            >>> import ray
-            >>> # Group by a table column and aggregate.
-            >>> ray.data.from_items([
-            ...     {"A": x % 3, "B": x} for x in range(100)]).groupby(
-            ...     "A").count()
-            Aggregate
-            +- Dataset(num_blocks=..., num_rows=100, schema={A: int64, B: int64})
+
+            .. testcode::
+
+                import pandas as pd
+                import ray
+
+                def transform_variety(group: pd.DataFrame) -> pd.DataFrame:
+                    ...  # Separately transform each variety
```

Comment:
can we add an actual example here? It's not clear what I should expect the grouped data to look like